compiler/optimizations/optimizeOnClauses.cpp

```diff
@@ -125,19 +141,34 @@ isFastPrimitive(CallExpr *call, bool isLocal) {
   case PRIM_AND_ASSIGN:
   case PRIM_OR_ASSIGN:
   case PRIM_XOR_ASSIGN:
-    if (!isCallExpr(call->get(2))) { // callExprs checked in calling function
-      if (!call->get(1)->typeInfo()->symbol->hasFlag(FLAG_WIDE_REF) &&
-          !call->get(2)->typeInfo()->symbol->hasFlag(FLAG_WIDE_REF)) {
-        DEBUG_PRINTF(" *** OK (PRIM_MOVE 1): %s\n", call->primitive->name);
-        return true;
-      }
-    } else {
-      DEBUG_PRINTF(" *** OK (PRIM_MOVE 2): %s\n", call->primitive->name);
+    if (isCallExpr(call->get(2))) { // callExprs checked in calling function
       // Not necessarily true, but we return true because
       // the callExpr will be checked in the calling function
-      return true;
+      return FAST_AND_LOCAL;
+    } else {
+      bool arg1wide = call->get(1)->typeInfo()->symbol->hasFlag(FLAG_WIDE_REF);
```

Comment:
optimizeOnClauses, at least in its current form, couldn't possibly work before insertWideReferences as far as I know. Since most of the logic here is checking if the argument is a wide pointer or not (e.g. in classifyPrimitive). Also, for historical background, it used to be in parallel but made it to its own pass (for whatever reason).

There may be some obvious way to make this pass work before insertWideReferences but I don't see what it is.